tests/expressions/tests.py

```diff
@@ -331,6 +331,101 @@ def test_object_update_unsaved_objects(self):
         with self.assertRaisesMessage(ValueError, msg):
             acme.save()
 
+    def test_ticket_22288_in_f_integer(self):
+        # Test that in lookups can use F() expressions for integers
+        ceo = Employee.objects.create(firstname="Just", lastname="Doit", salary=30)
+        Company.objects.create(name="5020 Ltd", num_employees=50, num_chairs=20, ceo=ceo)
+        Company.objects.create(name="5040 Ltd", num_employees=50, num_chairs=40, ceo=ceo)
+        Company.objects.create(name="5050 Ltd", num_employees=50, num_chairs=50, ceo=ceo)
+        Company.objects.create(name="5060 Ltd", num_employees=50, num_chairs=60, ceo=ceo)
+        Company.objects.create(name="99300 Ltd", num_employees=99, num_chairs=300, ceo=ceo)
+
+        queryset = Company.objects.filter(
+            num_employees__in=([F('num_chairs') - 10])
+        )
+        self.assertQuerysetEqual(queryset, ['<Company: 5060 Ltd>'], ordered=False)
+
+        queryset = Company.objects.filter(
+            num_employees__in=([F('num_chairs') - 10, F('num_chairs') + 10])
+        )
+        self.assertQuerysetEqual(queryset, ['<Company: 5040 Ltd>',
+                                            '<Company: 5060 Ltd>'], ordered=False)
+
+    def test_ticket_22288_range_f_integer(self):
+        # Test that range lookups can use F() expressions for integers
+        ceo = Employee.objects.create(firstname="Just", lastname="Doit", salary=30)
+        Company.objects.create(name="5020 Ltd", num_employees=50, num_chairs=20, ceo=ceo)
+        Company.objects.create(name="5040 Ltd", num_employees=50, num_chairs=40, ceo=ceo)
+        Company.objects.create(name="5050 Ltd", num_employees=50, num_chairs=50, ceo=ceo)
+        Company.objects.create(name="5060 Ltd", num_employees=50, num_chairs=60, ceo=ceo)
+        Company.objects.create(name="99300 Ltd", num_employees=99, num_chairs=300, ceo=ceo)
+
+        Company.objects.filter(num_employees__exact=F("num_chairs"))
+
+        queryset = Company.objects.filter(num_employees__range=(F('num_chairs'), 100))
+        self.assertQuerysetEqual(queryset, ['<Company: Test GmbH>',
```

Comment:
When updating the tests, please use the indentation style of the existing tests, e.g.
```python
self.assertQuerysetEqual(
    Number.objects.all(),
    [
        '<Number: -1, -1.000>',
        '<Number: 43, 42.000>',
        '<Number: 1338, 1337.000>'
    ],
    ordered=False
)
```